src/cursor.rs

```diff
@@ -0,0 +1,116 @@
+use std::ops::{Add, AddAssign, Index, IndexMut, Sub, SubAssign};
+
+// TODO: AddAssign
+pub struct CursorMut<'a, T> {
+    data: &'a mut [T],
```

Comment:
What about making this unsized, so it's like an even fatter slice?  Then either `&` or `&mut` can be added to it so it's more flexible, and I think `Clone` should work normally (not sure).  I'm not sure if this would work, but if it does I think it'd be cleaner.

Reply:
I don't think that would work? I thought there wasn't a way to construct those DSTs currently? At least that's what the nomicon says, idk if it's out of date. But I think it's probably best to avoid using DSTs because language support for them is kind of lacking, and that's going to add maintenance complexity that we don't need.

Reply:
Yeah, you're right.  I didn't realize constructing DSTs was still so unstable.  That's unfortunate.